src/renderer/utils/words.ts

```diff
@@ -15,10 +16,22 @@ export const isInOriginalOrder: (
       return true;
     }
 
+    // edge case, only one word
+    if (getLengthOfRange(range) === 1 || words.length === 1) {
+      return true;
+    }
+
+    const prevWordOriginalIndex =
```

Comment:
These changes aren't strictly required but I realised there was a small bug in this function that was getting hit because of the bugs in the other stuff, so I fixed that as well